pytorch_to_returnn/torch/tensor.py

```diff
@@ -122,10 +122,24 @@ def float(self):
     return cast(self, "float32")
 
   def __getitem__(self, item):
-    assert isinstance(item, int)  # not implemented otherwise
     assert self._shape  # cannot subscript a scalar
-    from .nn import Gather
-    return Gather(dim=0, pos=item)(self)
+    if isinstance(item, int):
+      from .nn import Gather
+      return Gather(dim=0, pos=item)(self)
+    elif isinstance(item, slice):
+      from .nn import Slice
+      return Slice(axis=0, start=item.start, stop=item.stop, step=item.step)(self)
+    elif isinstance(item, tuple):
+      assert len(item) == self.ndim
+      from .nn import Slice
+      out = None
+      for ax, ax_slice in enumerate(item):
+        assert isinstance(ax_slice, slice)
+        out = Slice(axis=ax, start=ax_slice.start, stop=ax_slice.stop, step=ax_slice.step)(out or self)
```

Comment:
```suggestion
        out = Slice(axis=ax, start=ax_slice.start, stop=ax_slice.stop, step=ax_slice.step)(out)
```
(Always try to write code as simple as possible, and as least error-prone as possible, with as least branching (`if`, `or`, ...) as possible.)

Reply:
You can make one other optimization though:
```
if not ax_slice.start and not ax_slice.stop and ax_slice.step in {1, None}:
  continue
```
This will look nicer in the resulting RETURNN net.

Reply:
Sure, I'll do that. But I don't think this will happen often, will it?

Reply:
This happens exactly in your test case. I think this is quite common.

Reply:
Oh yes, it's the single colon. I see.